File: JavaScriptPractice/Leetcode2022.js
var construct2DArray = function(original, m, n) 
{
    let result=[];
    let count=0;
    for (let i = 0; i < m; i++) {
        let row=[];
        for (let j = 0; j < n; j++) {
            row.push(original[count]);
            count++;
            
        }
        result.push(row);
        
    }
    return result;
    
};

File: CppPractice/CppLeetcode/Leetcode2022.cpp
#include<vector>
using namespace std;

class Solution {
public:
    vector<vector<int>> construct2DArray(vector<int>& original, int m, int n) 
    {
        if (m*n!= original.size())
        {
            return vector<vector<int>>();
        }
        int count = 0;
        vector<vector<int>> ans;
        for (int i = 0; i < m; i++)
        {
            vector<int> temp;
            for (int j = 0; j < n; j++)
            {
                temp.push_back(original[count]);
                count++;
            }
            ans.push_back(temp);
        }
        return ans;
        
        
    }
};